### PR TITLE
Support bors try arguments.

### PR DESCRIPTION
**Pull request by reitermarkus**
Allow trying individual targets.

Closes https://github.com/cross-rs/cross/issues/651.

### CODE REVIEW

**Comment by reitermarkus** [comment]
bors try --target dummy

**Comment by Emilgardis** [comment]
if only there was a way to actually debug events properly :/

**Comment by Emilgardis** [comment]
bors try --target dummy_.*

**Comment by reitermarkus** [comment]
bors try --target "dummy"

**Comment by reitermarkus** [comment]
bors try --target "dummy"

**Comment by reitermarkus** [comment]
bors try --target "dummy"

**Comment by reitermarkus** [comment]
bors try --target dummy_.*

**Comment by reitermarkus** [comment]
bors try-

**Comment by reitermarkus** [comment]
bors try --target dummy_.*

**Comment by reitermarkus** [comment]
bors try --target dummy_.*

**Comment by Emilgardis** [comment]
failing on a non-matched target instead of trying to correct the matrix is what should be done imo. Don't think it's possible to do conditional matrices

**Comment by reitermarkus** [comment]
I'm not quite sure what you mean.

**Comment by reitermarkus** [comment]
bors try --target freebsd

**Comment by reitermarkus** [comment]
bors try-

**Comment by reitermarkus** [comment]
bors try --target freebsd

**Comment by reitermarkus** [comment]
bors try-

**Comment by reitermarkus** [comment]
bors try --target freebsd

**Comment by Dylan-DPC** [comment]
i have a feeling we need to merge this and then test it out on a separate PR

**Comment by reitermarkus** [comment]
bors try --target freebsd

**Comment by Emilgardis** [comment]
pseudo code

```
    steps:
      - uses: actions/checkout@v3
	  - uses: get-targets-from-bors-try
	    if: <branch> == trying
      - name: skip if trying and not matching
        if: <branch> == trying && !contains(steps.get-targets-from-bors.outputs.target-list, matrix.target)
      - uses: ./.github/actions/setup-rust
      - name: Set up Docker Buildx
        if: runner.os == 'Linux'
```



> i have a feeling we need to merge this and then test it out on a separate PR

can create a test repo for it

**Comment by reitermarkus** [comment]
Basically what I want to do is filter the targets from the default matrix and only run those. It's definitely possible to dynamically generate the matrix, but I'll continue trying this tomorrow.

**Comment by reitermarkus** [comment]
bors try --target freebsd

**Comment by reitermarkus** [comment]
bors try --target freebsd

**Comment by reitermarkus** [comment]
bors try-

**Comment by reitermarkus** [comment]
bors try --target freebsd

**Comment by reitermarkus** [comment]
bors try --target freebsd

**Comment by reitermarkus** [comment]
bors try-

**Comment by reitermarkus** [comment]
bors try --target freebsd

**Comment by Emilgardis** [comment]
This is great, did not realise this was possible 👍

**Comment by reitermarkus** [comment]
bors try --target powerpc

**Comment by reitermarkus** [comment]
bors try

**Comment by reitermarkus** [comment]
Disabled the GHA cache since it fails sporadically and isn't big enough anyways to hold all images.

**Comment by reitermarkus** [comment]
bors try

**Comment by reitermarkus** [comment]
bors try

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+- #652 - Allow trying individual targets via bors.
```
Comment (Emilgardis) [inline CHANGELOG.md:8]: can you add 609 to the changelog also :)

**Review by Emilgardis — APPROVED** [review]
r=me, small nit not related to the pr, can be done separately

**Pull request merged**